solana/program_v2/programs/gateway_v2/src/state/network.rs

```diff
@@ -99,44 +99,25 @@ impl GatekeeperNetwork {
             > 0
     }
 
-    pub fn set_expire_time(
-        &mut self,
-        data: &UpdateNetworkData,
-        authority: &mut Signer,
-    ) -> Result<()> {
-        match data.pass_expire_time {
-            Some(pass_expire_time) => {
-                if pass_expire_time != self.pass_expire_time {
-                    if !self.can_access(authority, NetworkKeyFlags::SET_EXPIRE_TIME) {
-                        return Err(error!(NetworkErrors::InsufficientAccessExpiry));
-                    }
-
-                    self.pass_expire_time = pass_expire_time;
-                }
-
-                Ok(())
+    pub fn set_expire_time(&mut self, pass_expire_time: Option<i64>) -> Result<()> {
+        if let Some(pass_expire_time) = pass_expire_time {
```

Comment:
Sorry this is out of the scope, but why is the pass_expire_time an `Option`? Because it could NOT have an expiry time?

Should I not then set the expire time to `None`?

Reply:
On an update, if no pass expiry is provided - then don't change it.

Based on your earlier feedback, it would be better to do that check outside the function and not be an `Option` here right?

Reply:
Why would you call `set_expire_time` if you don't want to set the expire time? :)

Reply:
Think it was just part of the update_network function.

```
pub fn update_network(ctx: Context<UpdateNetworkAccount>, data: &UpdateNetworkData) -> Result<()> {
    let network = &mut ctx.accounts.network;
    let authority = &mut ctx.accounts.authority;

    **network.set_expire_time(data, authority)?;**
    network.update_auth_keys(data, authority)?;
    network.update_fees(data, authority)?;
    network.update_network_features(data, authority)?;
    network.update_supported_tokens(data, authority)?;

    Ok(())
}
```
Maybe should be separated so that functions only run if the corresponding field in the data struct has a valid entry

Reply:
>Why would you call set_expire_time if you don't want to set the expire time? :)

It's the blanket update function... btw, after this - not a fan of these big update functions at all :)

Reply:
i.e. in the overall update, providing a new expire time is optional from the client